src/EnergyPlus/AirflowNetwork/src/Elements.cpp

```diff
@@ -0,0 +1,2994 @@
+    Array1D<AirflowNetworkLinkSimuData> AirflowNetworkLinkSimu;
+    Array1D<AirflowNetworkExchangeProp> AirflowNetworkExchangeData;
+    Array1D<AirflowNetworkExchangeProp> AirflowNetworkMultiExchangeData;
+    Array1D<AirflowNetworkLinkReportData> AirflowNetworkLinkReport;
+    Array1D<AirflowNetworkNodeReportData> AirflowNetworkNodeReport;
+    Array1D<AirflowNetworkLinkReportData> AirflowNetworkLinkReport1;
+    AirflowNetworkSimuProp AirflowNetworkSimu;
+    Array1D<AirflowNetworkNodeProp> AirflowNetworkNodeData;
+    Array1D<AirflowNetworkCompProp> AirflowNetworkCompData;
+    Array1D<AirflowNetworkLinkageProp> AirflowNetworkLinkageData;
+    Array1D<MultizoneZoneProp> MultizoneZoneData;
+    Array1D<MultizoneSurfaceProp> MultizoneSurfaceData;
+    Array1D<DetailedOpening> MultizoneCompDetOpeningData;
+    Array1D<SimpleOpening> MultizoneCompSimpleOpeningData;
+    Array1D<HorizontalOpening> MultizoneCompHorOpeningData;
+    // Array1D<ReferenceConditions> MultizoneSurfaceStdConditionsCrackData;
+    Array1D<SurfaceCrack> MultizoneSurfaceCrackData;
+    Array1D<EffectiveLeakageArea> MultizoneSurfaceELAData;
+    Array1D<MultizoneExternalNodeProp> MultizoneExternalNodeData;
+    Array1D<DeltaCpProp> DeltaCp;
+    Array1D<DeltaCpProp> EPDeltaCP;
+    Array1D<ZoneExhaustFan> MultizoneCompExhaustFanData;
+    Array1D<IntraZoneNodeProp> IntraZoneNodeData;
+    Array1D<IntraZoneLinkageProp> IntraZoneLinkageData;
+    Array1D<DisSysNodeProp> DisSysNodeData;
+    Array1D<DuctLeak> DisSysCompLeakData;
+    Array1D<EffectiveLeakageRatio> DisSysCompELRData;
+    Array1D<Duct> DisSysCompDuctData;
+    Array1D<Damper> DisSysCompDamperData;
+    Array1D<ConstantVolumeFan> DisSysCompCVFData;
+    Array1D<DetailedFan> DisSysCompDetFanData;
+    Array1D<DisSysCompCoilProp> DisSysCompCoilData;
+    Array1D<DisSysCompHXProp> DisSysCompHXData;
+    Array1D<DisSysCompTermUnitProp> DisSysCompTermUnitData;
+    Array1D<ConstantPressureDrop> DisSysCompCPDData;
+    Array1D<AiflowNetworkReportProp> AirflowNetworkReportData;
+    Array1D<PressureControllerProp> PressureControllerData;
+    Array1D<OutdoorAirFan> DisSysCompOutdoorAirData;
+    Array1D<ReliefFlow> DisSysCompReliefAirData;
+    Array1D<AirflowNetworkLinkageViewFactorProp> AirflowNetworkLinkageViewFactorData;
+
+    static Real64 square(Real64 x)
+    {
+        return x * x;
+    }
+
+    int Duct::calculate(bool const LFLAG,           // Initialization flag.If = 1, use laminar relationship
+                        Real64 const PDROP,         // Total pressure drop across a component (P1 - P2) [Pa]
+                        int const EP_UNUSED(i),     // Linkage number
+                        const AirProperties &propN, // Node 1 properties
+                        const AirProperties &propM, // Node 2 properties
+                        std::array<Real64, 2> &F,   // Airflow through the component [kg/s]
+                        std::array<Real64, 2> &DF   // Partial derivative:  DF/DP
```

Comment:
Would it be beneficial for these two arrays to be const refs?

Reply:
No, those are two of the three returns out of these functions. I thought about wrapping all three into a struct, but I'm not quite sold on that yet.

Reply:
:+1: